hedyorg/hedy
# 🪲 Add function definitions to the global scope

This PR corrects an issue with function definitions appearing only in local scopes in the lookup table. As a result, calling one function with arguments from another raised an error.

Fixes #5983

**How to test**
- an automated tests is added
- the following code should execute in level 15
```
old_sequence = 'test', 'test'
new_sequence = 'test', 'test'

define add_item with number
    item = ask 'What is item number ' number '?'
    add item to new_sequence
    print item

define show_correct_answer
    print 'The correct items are:'
    for item in old_sequence
        if item != 'test'
            print item

define get_player_sequence with length
    i = 1
    while i < length
        call add_item with i
        i = i + 1
    if new_sequence == old_sequence
        call add_item with i
        print "Correct! Moving on to the next level."
        correct = 'True'
    else
        print 'Incorrect! Game Over.'
        call show_correct_answer
        correct = 'False'
    return correct

correct = 'True'
level = 1
player = 1
while correct == 'True'
    sleep 1
    clear
    print 'Player ' player
    print 'Ik ga op reis en neem mee'
    correct = call get_player_sequence with level
    level = level + 1
    old_sequence = new_sequence
    new_sequence = 'test', 'test'
    if player == 1
        player = 2
    else
        player = 1
```
